SimPEG/dask/electromagnetics/static/induced_polarization/simulation.py

```diff
@@ -1,22 +1,23 @@
+import dask.array as da
+
 from .....electromagnetics.static.induced_polarization.simulation import (
     BaseIPSimulation as Sim,
 )
 
-import dask.array as da
-
 
 def dask_getJtJdiag(self, m, W=None, f=None):
     """
     Return the diagonal of JtJ
     """
     if getattr(self, "_gtgdiag", None) is None:
+        J = self.getJ(m, f=f)
         # Need to check if multiplying weights makes sense
         if W is None:
-            W = self._scale
+            W = self.scale
         else:
-            W = self._scale * W.diagonal()
+            W = self.scale * W.diagonal()
         w = da.from_array(W)[:, None]
-        self._gtgdiag = da.sum((w * self.getJ(m, f=f)) ** 2, axis=0).compute()
+        self._gtgdiag = da.sum((w * J) ** 2, axis=0).compute()
```

Comment:
```suggestion
        self._gtgdiag = da.einsum('i,ij,ij->j', W**2., self.Jmatrix, self.Jmatrix)
```

Reply:
Not sure if the current setup knows about Jmatrix, but at least einsum should be much faster than `(W * J) ** 2.`